features/configuration.feature

```diff
@@ -0,0 +1,89 @@
+Feature: Configure integration with environment variables
+
+    The library should be configurable using environment variables to support
+    single-line and reusable configuration
+
+    Background:
+        Given I set environment variable "BUGSNAG_API_KEY" to "a35a2a72bd230ac0aa0f52715bbdc6aa"
+        And I configure the bugsnag endpoint
+        And I have built the service "autoconfigure"
+
+    Scenario Outline: Adding content to handled events through env variables
+        Given I set environment variable "<variable>" to "<value>"
+        And I set environment variable "BUGSNAG_AUTO_CAPTURE_SESSIONS" to "0"
+        When I run the go service "autoconfigure" with the test case "<testcase>"
+        Then I wait to receive a request
+        And the request is a valid error report with api key "a35a2a72bd230ac0aa0f52715bbdc6aa"
+        And the event "<field>" equals "<value>"
+
+        Examples:
+            | testcase | variable                              | value           | field                         |
+            | panic    | BUGSNAG_APP_VERSION                   | 1.4.34          | app.version                   |
+            | panic    | BUGSNAG_APP_TYPE                      | mailer-daemon   | app.type                      |
+            | panic    | BUGSNAG_RELEASE_STAGE                 | beta1           | app.releaseStage              |
+            | panic    | BUGSNAG_HOSTNAME                      | dream-machine-2 | device.hostname               |
+            | panic    | BUGSNAG_METADATA_device_instance      | kube2-33-A      | metaData.device.instance      |
+            | panic    | BUGSNAG_METADATA_framework_version    | v3.1.0          | metaData.framework.version    |
+            | panic    | BUGSNAG_METADATA_device_runtime_level | 1C              | metaData.device.runtime level |
+            | panic    | BUGSNAG_METADATA_Carrot               | orange          | metaData.custom.Carrot        |
```

Comment:
Is setting metadata this way definitely something we want to support? I don't think we do anywhere else

Reply:
yeah. no code configuration might be necessary for future changes, though for now I've split this into a separate changeset.